docs/source/index.rst

```diff
@@ -74,6 +68,15 @@ Welcome to TensorRT-LLM's Documentation!
    1.0/features/speculative-decoding.md
 
 
+.. toctree::
+   :maxdepth: 2
+   :caption: Developer Guide
+
+   1.0/architecture/overview.md
```

Comment:
I just noticed that the new doc files are placed in `1.0/` directory, is that intended for faster iterations before 1.0? And will we remove that directory before the 1.0 publish?